### ISSUE TITLE
Train data and image are different

### ISSUE THREAD

**Issue by manu123416** [mit-han-lab/torchquantum#40]
Hi ,
I tried testing your quantum neural network code on jupyter notebook.
I think, there is some bug in the training data.
```
dataset = MNIST(root='../Data_Manu',
                 train_valid_split_ratio=[0.9, 0.1],
            digits_of_interest=[3, 5],
#             n_train_samples = 75,
            n_test_samples=75)
```

data_train= dataset['train'][0]

```
{'image': tensor([[[-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.3733, -0.1696,
           -0.1696,  0.8868,  1.4468,  2.1087,  0.0213, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242,  0.1995,  0.6704,  1.8032,  1.9560,  2.7960,
            2.8088,  2.7960,  2.7960,  2.7960,  2.4142, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
            1.6250,  2.1723,  1.3068,  1.3068,  1.3196,  1.3068,  1.3068,
            1.4978,  2.5542,  2.7069,  2.7960,  2.7960,  2.7960,  2.7960,
            2.8088,  2.7960,  2.7960,  2.3251,  1.8160, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
            2.2996,  2.7960,  2.7960,  2.7960,  2.8088,  2.7960,  2.7960,
            2.7960,  2.7960,  2.8088,  2.7960,  2.7960,  2.7960,  2.7960,
            2.0323,  0.9886,  0.3140, -0.3606, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
            0.7977,  2.8088,  2.8088,  2.8088,  2.8215,  2.8088,  2.8088,
            2.8088,  2.8088,  2.6433,  1.9560,  0.8232,  0.6322, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
            0.5049,  2.7960,  2.7960,  2.7960,  2.8088,  2.4778,  1.2941,
            0.6322,  0.0722, -0.0424, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,  0.4795,
            2.6433,  2.7960,  2.7960,  2.7960,  1.0523, -0.2715, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,  2.2360,
            2.7960,  2.7960,  2.5160,  0.5813, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,  2.8088,
            2.7960,  2.7960,  2.4906,  0.8232,  0.8359,  0.8232,  0.8232,
            0.8232, -0.1315, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,  2.0451,
            2.8088,  2.8088,  2.8088,  2.8088,  2.8215,  2.8088,  2.8088,
            2.8088,  2.8088,  2.0451,  0.1740, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,  0.1740,
            1.8796,  2.5415,  2.5415,  2.6433,  2.5542,  2.5415,  2.5415,
            2.5415,  2.6433,  2.8088,  2.3760, -0.2460, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.0424, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.0424,  2.4142,  2.7960,  2.1087, -0.3478, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.2206,  2.2360,  2.7960,  2.7960, -0.1824, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.0296,
            1.4850,  2.3378,  2.8088,  2.7960,  2.7960,  0.3013, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242,  0.6195,  1.6505,  2.8088,
            2.8088,  2.8088,  2.8215,  2.8088,  1.9051, -0.3224, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.3351,  0.5049,  2.0196,  2.8088,  2.7960,  2.7960,
            2.7960,  2.7960,  2.4524,  1.2050, -0.2715, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.3351,  1.7269,  2.7960,  2.7960,  2.8088,  2.7960,  2.7960,
            2.6306,  1.7905,  0.3395, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.3097,  1.4341,  2.2869,  2.2869,  2.2996,  1.7141,  1.0650,
           -0.1315, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242],
          [-0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242,
           -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242, -0.4242]]]),
 'digit': 1}
```

```
The tensor matrix contains 5 but the label shows 1?

**Comment by Hanrui-Wang** [comment]
Hi manu123416,

As shown here: https://github.com/mit-han-lab/torchquantum/blob/772f0f78a7db50c9f09c866f5bc9e20fed7066c8/torchquantum/datasets/mnist.py#L194 the real digits will be converted to relative digit. This will make it more convenient for cross entropy loss training

**Comment by Hanrui-Wang** [comment]
For more data, could you try change the n_train_samples = 75 and see whether more data are used in training set?

**Comment by manu123416** [comment]
Thank you Hanui-Wang. I tested it by putting n_train_samples = 75, but it doesnot increase the the data the training and reduces the training data to 1.

**Comment by manu123416** [comment]
Also, I tried plotting the loss at the end of each epochs.
The validation results are like fixed as shown below.
![image](https://user-images.githubusercontent.com/95257431/197080769-230e1410-07da-411f-8a2f-5457af061654.png)
Is there a way I can share my jupyter notebook with you?
